src/shared/src/backend-supports/project-supports.ts

```diff
@@ -18,6 +18,16 @@ const calculateEndDate = (start: Date, weeks: number) => {
   return end;
 };
 
+/**
+ * This function calculates the end date for a project.
+ * @param wps an array of work packages
+ * @returns the latest end date of the workpackages
+ */
+const calculateProjectEndDate = (wps: { endDate: Date }[]) => {
```

Comment:
the endDate: Date part breaks my code, not sure why as the functions below use similar syntax. I narrowed it down to endDate being the issue but not sure why the endDate field breaks it.

Reply:
it's broken in the project transformer because work packages don't start out with an endDate field, it needs to be calculated like in my picture below: although I'm not sure this is great design. i think the function should take in a list of workpackages then (or list of objects with {startDate: Date; duration: number} and then go from there so that the logic is most contained to within the util function

![Screen Shot 2022-10-06 at 10 22 25 AM](https://user-images.githubusercontent.com/63663597/194338274-ffc1180f-1645-4476-a764-e6321e578698.png)